### ISSUE TITLE
WHERE clause is missing, if a left join is done on a referenced property

### ISSUE THREAD

**Issue by mhobl** [dotnet/efcore#5834]
### Steps to reproduce

I have the following setup:

```
    public class Activity
    {
        public string ActivityId { get; set; }
        public string UserDevicePairId { get; set; }
        public UserDevicePair UserDevicePair { get; set; }
    }

 public class UserDevicePair
    {
        public string UserDevicePairId { get; set; }
        public string ApplicationUserId { get; set; }
        public ApplicationUser ApplicationUser { get; set; }
        public ICollection<Activity> Activities { get; set; }
    }

 public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
    {
        public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
            : base(options)
        {
        }

        protected override void OnModelCreating(ModelBuilder builder)
        {
            base.OnModelCreating(builder);
            builder.Entity<ApplicationUser>().HasKey(u => u.Id);
            builder.Entity<UserDevicePair>().HasKey(a => a.UserDevicePairId);
            builder.Entity<UserDevicePair>()
                .HasOne(s => s.ApplicationUser)
                .WithMany(u => u.UserDevicePairs)
                .OnDelete(DeleteBehavior.Restrict);
            builder.Entity<Activity>().HasKey(s => s.ActivityId);
            builder.Entity<Activity>()
                .HasOne(a => a.UserDevicePair)
                .WithMany(u => u.Activities)
                .HasForeignKey(u => u.UserDevicePairId)
                .OnDelete(DeleteBehavior.Restrict);
        }
        public DbSet<Activity> Activities { get; set; }
        public DbSet<UserDevicePair> UserDevicePairs { get; set; }
    }
```

and this repository

```
 public class ActivityRepository
    {
        private readonly ApplicationDbContext _context;
        public ActivityRepository(ApplicationDbContext dbcontext)
        {
            _context = dbcontext;
        }
        public async Task<IEnumerable<Activity>> GetAsync(string userId)
        {
            var activities = await _context.Activities
                .Where(s => s.UserDevicePair.ApplicationUserId == userId)
                .ToListAsync();

            return activities;
        }
    }
```
### The issue

The generated SQL statement looks like this one:

```
SELECT [s].[ActivityId], [s].[UserDevicePairId], [s.UserDevicePair].[UserDevicePairId], [s.UserDevicePair].[ApplicationUserId]
FROM [Activities] AS [s]
LEFT JOIN [UserDevicePairs] AS [s.UserDevicePair] ON [s].[UserDevicePairId] = [s.UserDevicePair].[UserDevicePairId]
ORDER BY [s].[UserDevicePairId]
```

but I would expect the this SQL statement with the WHERE clause:

```
SELECT [s].[ActivityId], [s].[UserDevicePairId], [s.UserDevicePair].[UserDevicePairId], [s.UserDevicePair].[ApplicationUserId]
FROM [Activities] AS [s]
LEFT JOIN [UserDevicePairs] AS [s.UserDevicePair] ON [s].[UserDevicePairId] = [s.UserDevicePair].[UserDevicePairId]
WHERE [s.UserDevicePair].[ApplicationUserId] = '...'
ORDER BY [s].[UserDevicePairId]
```

EF does a left join on the complete table before selecting the right user.
### Further technical details

EF Core version: 1.0.0-rc2-final
Operating system: Win10
Visual Studio version: VS 2015

Other details about my project setup:

**Comment by pryankster** [comment]
wow.  I was JUST about to post this issue ... was packaging up a test case and putting it on GIT 
https://github.com/pryankster/EFTests

--pryankster

**Comment by mhobl** [comment]
Ping @rowanmiller @anpete 
A statement from a asp.net member would be nice, because this is a critical bug for me, as a can't do a WHERE statement on a hundred thousand to million datasets without waiting at least 30 - 40 seconds until the left join is done without any restrictions...

**Comment by rowanmiller** [comment]
This is a duplicate of https://github.com/aspnet/EntityFramework/issues/4588. Unfortunately, the fix did not get into 1.0.0. The issue is sitting in the 1.1.0 release, but given the impact it may get pulled into a patch release.

**Comment by rowanmiller** [comment]
Closing as this is tracked by #4588
